src/runtime/vdom/vdom-render.ts

```diff
@@ -418,24 +418,22 @@ const updateFallbackSlotVisibility = (elm: d.RenderNode) => {
         childNode.hidden = false;
 
         for (j = 0; j < ilen; j++) {
-          if (childNodes[j]['s-hn'] !== childNode['s-hn']) {
-            // this sibling node is from a different component
-            nodeType = childNodes[j].nodeType;
-
-            if (slotNameAttr !== '') {
-              // this is a named fallback slot node
-              if (nodeType === NODE_TYPE.ElementNode && slotNameAttr === childNodes[j].getAttribute('slot')) {
-                childNode.hidden = true;
-                break;
-              }
-            } else {
-              // this is a default fallback slot node
-              // any element or text node (with content)
-              // should hide the default fallback slot node
-              if (nodeType === NODE_TYPE.ElementNode || (nodeType === NODE_TYPE.TextNode && childNodes[j].textContent.trim() !== '')) {
-                childNode.hidden = true;
-                break;
-              }
+          // this sibling node is from a different component
```

Comment:
This comment was regarding the `if (childNodes[j]['s-hn'] !== childNode['s-hn'])` statement, so should also be deleted. Wondering whether this condition was important for some other edge cases though (which might also not have a test yet).

Maybe sth like

```ts
if (childNodes[j]['s-hn'] !== childNode['s-hn'] || slotNameAttr !== '') {
  // ...
}
```

would be better?